### PR TITLE
Add Mainnet to CA; add dropdown for CA network select to Settings (DAP-3018)

### PR DESCRIPTION
**Pull request by Ni-2**
#### What is the purpose of this pull request? (put an "X" next to an item)

[ ] Bug fix
[x] New feature
[ ] Documentation update
[ ] Add something to the core
[ ] Other, please explain:

#### Add a reference to a related issue if it exists

*issue:*

### CODE REVIEW

File: src/background/services/connectedAccountService.ts [hunk 1]
```diff
@@ -20,32 +21,54 @@ type EthSignature = {
...
+        if (!this._testnetContract) {
+          const contractAddress =
+            await this._globalConfigService.getConnectedAccountsTestnetContractAddress()
+          const near_account = await this._walletService.near_getAccount(
+            DefaultSigners.EXTENSION,
+            ChainTypes.NEAR_TESTNET
+          )
+          this._testnetContract = this._createContract(near_account, contractAddress)
+        }
+        return this._testnetContract
```
Comment (alsakhaev) [inline src/background/services/connectedAccountService.ts:70]: The same code. Need to be generalized.

File: src/background/services/globalConfigService.ts [hunk 2]
```diff
@@ -38,7 +39,9 @@ const EXPORTABLE_PROPERTIES = [
   'dynamicAdapter',
   'preferedOverlayStorage',
   'myDapplets',
-  'connectedAccountContractAddress',
+  'connectedAccountsTestnetContractAddress',
+  'connectedAccountsMainnetContractAddress',
```
Comment (alsakhaev) [inline src/background/services/globalConfigService.ts:43]: These hardcoded networks must be fixed in DAP-1552

File: src/background/services/globalConfigService.ts [hunk 3]
```diff
@@ -65,9 +68,9 @@ export default class GlobalConfigService {
       if (!config.myDapplets) config.myDapplets = this.getInitialConfig().myDapplets
       if (!config.targetStorages) config.targetStorages = this.getInitialConfig().targetStorages
       if (!config.xdaiProviderUrl) config.xdaiProviderUrl = this.getInitialConfig().xdaiProviderUrl
-      if (!config.connectedAccountsContractAddress)
-        config.connectedAccountsContractAddress =
-          this.getInitialConfig().connectedAccountsContractAddress
+      if (!config.preferredConnectedAccountsNetwork)
+        config.preferredConnectedAccountsNetwork =
+          this.getInitialConfig().preferredConnectedAccountsNetwork
```
Comment (alsakhaev) [inline src/background/services/globalConfigService.ts:73]: It seems that you forgot 'connectedAccountsTestnetContractAddress', 'connectedAccountsMainnetContractAddress'.

File: src/contentscript/core.ts [hunk 4]
```diff
@@ -537,4 +537,9 @@ export default class Core {
   public connectedAccounts = new ConnectedAccounts()
 
   public fetch = proxyFetch
+
+  public async getPreferredConnectedAccountsNetwork() {
```
Comment (alsakhaev) [inline src/contentscript/core.ts:541]: Should dapplets know about it?

**Review by alsakhaev — CHANGES_REQUESTED** [review]

File: src/background/services/connectedAccountService.ts (hunk 1 above)
Comment (alsakhaev) [reply src/background/services/connectedAccountService.ts:70]: For example
```typescrtipt
this._contracts[contractNetwork] = this._createContract(near_account, contractAddress)
```

File: src/contentscript/core.ts (hunk 4 above)
Comment (Ni-2) [reply src/contentscript/core.ts:541]: Yes. To understand what the network type use in the login to work with Connected Accounts.

File: src/background/services/globalConfigService.ts (hunk 3 above)
Comment (Ni-2) [reply src/background/services/globalConfigService.ts:79]: https://github.com/dapplets/dapplet-extension/commit/a95011495cc4ac2270ce8bc261e097f2ceb415f1
fixed

File: src/background/services/connectedAccountService.ts [hunk 5]
```diff
@@ -20,32 +21,54 @@ type EthSignature = {
 }
 
 export default class ConnectedAccountService {
-  _contract: any
+  _testnetContract: nearAPI.Contract
+  _mainnetContract: nearAPI.Contract
```
[the remark this answers is not in the document]
Comment (Ni-2) [reply src/background/services/connectedAccountService.ts:25]: add comment in https://dapplets-org.atlassian.net/browse/DAP-1552

File: src/background/services/globalConfigService.ts (hunk 2 above)
Comment (Ni-2) [reply src/background/services/globalConfigService.ts:43]: add comment in https://dapplets-org.atlassian.net/browse/DAP-1552

**Review by alsakhaev — APPROVED** [review]

**Pull request merged**